notebooks/403-action-recognition-webcam/403-action-recognition-webcam.ipynb

```diff
@@ -0,0 +1,646 @@
+    "    \"\"\"\n",
+    "    Decoder inference per set of frames. This function concatenates the embedding layer\n",
+    "    froms the encoder output, transpose the array to match with the decoder input size. \n",
+    "    Calls the network previously configured for the decoder model (exec_net_de), extracts\n",
+    "    the logits and normalize those to get confidence values along specified axis.\n",
+    "    Decodes top probabilities into corresponding label names \n",
+    "        \n",
+    "    :param: encoder_output: embedding layer for 16 frames\n",
+    "    :param: exec_net_de: Decoder model network\n",
+    "    :param: input_key_de: Input node network\n",
+    "    :returns: decoded_labels: The k most probable actions from the labels list\n",
+    "              decoded_top_probs: confidence for the k most probable actions\n",
+    "    \"\"\"\n",
+    "    # Concatenate sample_duration frames in just one array\n",
+    "    decoder_input = np.concatenate(encoder_output, axis=0)\n",
+    "    # Organize input shape vector to the Decoder (shape: [1x16x512]]\n",
+    "    decoder_input = decoder_input.transpose((2, 0, 1, 3))\n",
+    "    decoder_input = np.squeeze(decoder_input, axis=3)\n",
+    "\n",
+    "    # get results on action-recognition-0001-decoder model\n",
+    "    logits = exec_net_de.infer(inputs={input_key_de: decoder_input})\n",
+    "    result_de = logits[output_keys_de[0]]\n",
+    "    # Normalize logits to get confidence values along specified axis\n",
+    "    probs = softmax(result_de - np.max(result_de))\n",
+    "    # Decodes top probabilities into corresponding label names\n",
+    "    decoded_labels, decoded_top_probs = decode_output(probs, labels, top_k=3)\n",
+    "    return decoded_labels, decoded_top_probs\n",
+    "\n",
+    "\n",
+    "def softmax(x: np.array, axis: int = None) -> np.ndarray:\n",
+    "    \"\"\"\n",
+    "    Normalizes logits to get confidence values along specified axis\n",
+    "    \n",
+    "    x: np.array, axis=None\n",
+    "    \"\"\"\n",
+    "    exp = np.exp(x)\n",
+    "    return exp / np.sum(exp, axis=axis)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    },
+    "tags": []
+   },
+   "source": [
+    "### Main Processing Function\n",
+    "\n",
+    "Run action recognition function will run in different operations, either a webcam or a video file. See the list of procedures below:\n",
+    "\n",
+    "1. Create a video player to play with target fps (`utils.VideoPlayer`).\n",
+    "2. Prepare a set of frames to be encoded-decoded.\n",
+    "3. Run AI functions\n",
+    "4. Visualize the results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def run_action_recognition(source: str = 0, flip: bool = False, use_popup: bool = False, skip_first_frames: int = 0):\n",
```

Comment:
Should flip be True by default? I'm expecting most people will use self-facing webcams for this?